packages/hooks-react/src/useEntitlement.ts

```diff
@@ -1,11 +1,14 @@
-import { useQueries } from 'react-query';
+import { useMemo } from 'react';
+import { useQueries, useQuery } from 'react-query';
 import type { PlaylistItem } from '@jwp/ott-common/types/playlist';
 import type { GetEntitlementsResponse } from '@jwp/ott-common/types/checkout';
 import type { MediaOffer } from '@jwp/ott-common/types/media';
 import { getModule } from '@jwp/ott-common/src/modules/container';
 import { useConfigStore } from '@jwp/ott-common/src/stores/ConfigStore';
 import { useAccountStore } from '@jwp/ott-common/src/stores/AccountStore';
+import { useCheckoutStore } from '@jwp/ott-common/src/stores/CheckoutStore';
 import CheckoutController from '@jwp/ott-common/src/controllers/CheckoutController';
+import JWPEntitlementService from '@jwp/ott-common/src/services/JWPEntitlementService';
```

Comment:
This defeats the purpose of using an IoC Container. We should only import generic/abstract classes in the implementation.

Reply:
The service in question (`JWPEntitlementService`) was already used the same way in the `useContentProtection` hook: https://github.com/jwplayer/ott-web-app/blob/38f8a637489c497a07551450323dd21df0fc66a1/packages/hooks-react/src/useContentProtection.ts#L6

Since this PR is already big as it is and its scope isn't re-evaluating that implementation, I thought I'd just go along with it, because the contrary would've meant refactoring it with the possibility of breaking it.

Reply:
Ah that's right. That one isn't generalised yet unfortunately. 

I do wonder why it needs the additional access check? Isn't this covered by the `CheckoutService#getEntitlements` call?

Reply:
The problem is that offers are standalone entities, while plans are just containers of prices. But the access (or entitlement) is attached to the plan, rather than to the price. For this reason the mapping between offers and plans isn't consistent throughout the code - somewhere offer means plan, somewhere offer means plan price. So, one way of checking for access is iterating through every plan, making a request for each of them until we find which plan we have access for, which is what `getEntitlements` would do if we utilised it for the plans implementation. A quicker and more optimised way for this is to simply make a request to a different endpoint with `mediaId` and let the backend do the said iteration through the plans and then let us know if we have access for any of them. It's confusing, even more so with the constraints that go along with the need for generalisation.

Reply:
Thanks for the explanation. That does make things clear 😄

But I think that I still prefer to refactor the `getEntitlements` to make it fit the new APIs. Perhaps we should draw this out, but I don't have access to all the new APIs.

One thing that comes to mind is that instead of asking for entitlements for offer with id X, we could pass the full playlist item to the `getEntitlements` function. The service can then use whatever information it needs to figure out if the user has access.

Reply:
I understand what you mean and I would've definitely done that by now, but what we have is two totally different react queries with different variables. Just their query keys are indicative of that, where on the one hand we have `['token', 'media', playlistItem?.mediaid, {}]`, and on the other we have an array of queries with keys `['entitlements', offerId]`. Trying to reconcile them into a unifying one seems like too far beyond the scope of this feature - it would mean refactoring the existing implementation with the risk of breaking it, and I'm not even sure I understand why it has to be an array of `['entitlements', playlistItem.mediaOffers[i]]` query keys instead of a single one with `['entitlements', playlistItem.mediaid]`.

Reply:
I even forgot to bring up that the JWP integration still needs `getEntitlements` the way it is, it uses it from the `useCheckAccess` hook. I think we'll have to have two methods for checking for entitlements generically - `getEntitlementByAccessItem` (`access item` as a generic name for *offer* or *plan*) and `getEntitlementByMedia`, but I would leave it for a separate task.